### PR TITLE
Verify statedb

### PR DESCRIPTION
**Pull request by ethan-kr**
## Proposed changes

- This PR is derived from https://github.com/ethereum/go-ethereum/pull/21724
- Added snapshot subcommand in order to verify snapshot data
- It creates a trie based on snapshot data and compare the created root hash with the root hash in the header
- Added snapshot subcommand in order to trace statetrie nodes
- It reads all the nodes in the statetrie, looking for any missing nodes.
- Added iteration tools for snapshot nodes & statetrie nodes

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [x] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

#1229 

- https://github.com/klaytn/klaytn/pull/1555

### CODE REVIEW

File: cmd/utils/nodecmd/snapshot.go [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+		return errors.New("empty root")
+	}
+
+	logger.Info("Trace Start", "BlockNum", blockNumber)
+
+	sdb, err := state.New(root, state.NewDatabase(dbm), nil)
+	if err != nil {
+		return fmt.Errorf("Failed to open newDB trie : %v", err)
+	}
+	tridb := sdb.Database().TrieDB()
```
Comment (henry-will) [inline cmd/utils/nodecmd/snapshot.go:235]: Is triedb or trieDB better than tridb?

File: cmd/utils/nodecmd/snapshot.go [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+				utils.DynamoDBRegionFlag,
+				utils.DynamoDBIsProvisionedFlag,
+				utils.DynamoDBReadCapacityFlag,
+				utils.DynamoDBWriteCapacityFlag,
+				utils.LevelDBCompressionTypeFlag,
+				utils.DataDirFlag,
+			},
+			Description: `
+klaytn statedb iterate-triedb
+Coount the number of nodes in the state-trie db.
```
Comment (henry-will) [inline cmd/utils/nodecmd/snapshot.go:112]: Is it a typo in Count?

File: cmd/utils/nodecmd/snapshot.go [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+	}()
+
+	it := sdb.Database().TrieDB().DiskDB().GetStateTrieDB().NewIterator(nil, nil)
+	for it.Next() {
+		cnt++
+		if it.Key() == nil || it.Value() == nil || bytes.Equal(it.Key(), []byte("")) || bytes.Equal(it.Value(), []byte("")) {
+			nilCnt++
+		}
+	}
+	it.Release()
```
Comment (henry-will) [inline cmd/utils/nodecmd/snapshot.go:327]: In other code references, Iterator uses defer Release(). How about moving the 'defer it.Release()' statement to appear right after the 'it' variable is assigned.

**Review by henry-will — CHANGES_REQUESTED** [review]

File: cmd/kcn/main.go [hunk 4]
```diff
@@ -66,6 +66,9 @@ func init() {
 
 		// See utils/nodecmd/db_migration.go:
 		nodecmd.MigrationCommand,
+
+		// See utils/nodecmd/snapshot.go:
+		nodecmd.SnapshotCommand,
```
Comment (henry-will) [inline cmd/kcn/main.go:71]: SnapshotCommand are net required in kscn, kspn and ksen nodes, would it?

File: cmd/utils/nodecmd/snapshot.go (hunk 1 above)
Comment (ethan-kr) [reply cmd/utils/nodecmd/snapshot.go:235]: yes changed

File: cmd/utils/nodecmd/snapshot.go (hunk 2 above)
Comment (ethan-kr) [reply cmd/utils/nodecmd/snapshot.go:112]: Wow~~ chagned

File: cmd/kcn/main.go (hunk 4 above)
Comment (ethan-kr) [reply cmd/kcn/main.go:71]: Snapshot-related functions are not yet considered for the service chain.

**Review by jeongkyun-oh — APPROVED** [review]

**Review by henry-will — APPROVED** [review]

**Pull request merged**